cmd/controller/run.go

```diff
@@ -114,6 +114,26 @@ func Run(opts Options, runLog logr.Logger) error {
 	refTracker := reftracker.NewAppRefTracker()
 	updateStatusTracker := reftracker.NewAppUpdateStatus()
 
+	{ // add controller for config
+		reconciler := kcconfig.NewReconciler(coreClient, runLog.WithName("config"))
+
+		ctrl, err := controller.New("Config", mgr, controller.Options{
+			Reconciler: NewUniqueReconciler(&ErrReconciler{
```

Comment:
i dont think you want to be using `NewUniqueReconciler(&ErrReconciler{`, instead i think it should just be `reconciler`.

Reply:
I opted for this because it handles alot of the `synchronization` problems that could arise by processing the requests one at a time. Would changing it back to the plain reconciler not introduce the potential for race conditions on requests in parallel?

Reply:
MaxConcurrentReconciles=1 should take care of this.